src/sliding_sync.rs

```diff
@@ -951,26 +1032,47 @@ async fn async_main_loop(
     } else {
         None
     };
-
+    let cli: Cli = cli_parse_result.unwrap_or(Cli::default());
     let (client, _sync_token) = match new_login_opt {
         Some(new_login) => new_login,
-        None => loop {
-            log!("Waiting for login request...");
-            if let Some(login_request) = login_receiver.recv().await {
-                log!("Received login request for user {}", login_request.user_id);
-                match login(Cli::from(login_request)).await {
-                    Ok(new_login) => break new_login,
-                    Err(e) => {
-                        error!("Login failed: {e:?}");
-                        Cx::post_action(LoginAction::LoginFailure(e.to_string()));
-                        enqueue_rooms_list_update(RoomsListUpdate::Status {
-                            status: e.to_string(),
-                        });
+        None => {
+            let homeserver_url = cli.homeserver.clone()
+                .unwrap_or(String::from("https://matrix-client.matrix.org/"));
```

Comment:
`unwrap_or()` does eager evaluation; what you actually want here is deferred/lazy execution, so you should use `unwrap_or_else()`.
```suggestion
                .unwrap_or_else(|| String::from("https://matrix-client.matrix.org/"));
```

Reply:
Noted